src/matrixrtc/MatrixRTCSession.ts

```diff
@@ -205,6 +247,26 @@ export class MatrixRTCSession extends TypedEventEmitter<MatrixRTCSessionEvent, M
         return this.memberships[0];
     }
 
+    public onCallEncryption = (event: MatrixEvent): void => {
+        const userId = event.getSender();
+        const content = event.getContent<EncryptionKeyEventContent>();
+        const encryptionKey = content["m.encryption_key"];
+        const deviceId = content["m.device_id"];
+
+        if (
+            !userId ||
+            !deviceId ||
+            !encryptionKey ||
+            typeof deviceId !== "string" ||
+            typeof encryptionKey !== "string"
+        ) {
+            throw new Error("Malformed m.call.encryption_key");
+        }
+
+        this._encryptionKeys.set({ userId, deviceId }, encryptionKey);
```

Comment:
Also I would expect the encryption keys map to be initialised with anything the sdk already knows about when the session object is created, otherwise isn't there a risk we'll miss them?

Reply:
If we assume, we send new keys each time a new device joins the call, then no, we shouldn't miss them, right?

Reply:
Yeah, since we do this we can rely on that for now. Maybe this is good enough until we switch to todevice.